### DISCUSSION TITLE
How to Customize Link Colors and Highlight Text in Flowise AI Chatbot and Adjust the size of the image used in the chat button?

### DISCUSSION THREAD

**Discussion by ccamedia** [FlowiseAI/Flowise#2929] [Q&A] [answered]
Hi everyone,

I'm currently integrating a Flowise AI chatbot into my WordPress site using the Divi theme, and I'm looking to customize the appearance of links and highlighted text within the chat interface. Specifically, I want to:

Change the color of links to #fc9a3f.
Adjust the size of the image used in the chat button. I want to remove the space around and only use the picture as chat button. 

Thank you in advance for your help!

![link color](https://github.com/user-attachments/assets/fb5ce2e4-32bc-428d-9d8b-9297dfb8edaa)
![chat button](https://github.com/user-attachments/assets/6cb7dc59-82b4-4a51-ae5d-f5ab3e91ddd3)

**Comment by 714Lemons** [comment]
I think you have to change line 34 in this file: https://github.com/FlowiseAI/FlowiseChatEmbed/blob/main/src/assets/index.css
Here is some more information about the chat embed: https://docs.flowiseai.com/using-flowise/embed

**Reply by ccamedia** [reply]
Hi,

Thank you for your suggestion. Unfortunately, changing line 34 in the index.css file did not resolve the issue. The modifications to the link color and the size of the image in the chat bubble were not applied as expected.

Here are the steps I followed:

Cloned the repository from GitHub.
Edited the index.css file to update the link color and image size.
Ran pnpm install to install dependencies.
Executed pnpm build to build the project.
Despite these steps, the changes did not take effect. Is there something I might be missing or an alternative approach I could try?

![prose](https://github.com/user-attachments/assets/b3971a7b-ab9f-4841-b650-613734a434cf)
This is what I've found in the console...

**Reply by 714Lemons** [reply]
You are right it does not work. I looked into it and a fix is to add a line in the onMount function in BotBubble.tsx line 149:

onMount(() => {
    if (botMessageEl) {
      botMessageEl.innerHTML = Marked.parse(props.message.message);
      botMessageEl.querySelectorAll('a').forEach((link) => {
        link.target = '_self';
        // set color of the link to red
        link.style.color = '#ff0000';

**Reply by ccamedia** [reply]
Hi 714 Lemons, 
unfortunately, that solution did not work either. Despite adding the suggested line in the onMount function in BotBubble.tsx, the link color change did not take effect. Is there another approach or additional steps I should follow to resolve this issue? Any further guidance or a detailed tutorial would be greatly appreciated. this is my testingsite, where the bot is built in: cca-media.at/test/

**Reply by ccamedia** [reply]
![botbubble](https://github.com/user-attachments/assets/45430431-7be3-4f9f-a985-0bf14a97ca1a)

**Reply by ccamedia** [reply]
Hi @714Lemons, the links finally work, and the changes in the index.css are now effective. However, I'm struggling to change the colors of the headings and lists. I've added the styles to index.css using !important, but it doesn’t seem to have any effect. Any advice?
![css](https://github.com/user-attachments/assets/03b2a05b-b354-4590-acff-2f238641fae6)

**Reply by 714Lemons** [reply]
Hey I can't look at the code right now but I think it has something to do with the markdown rendering/parsing of the text. Maybe where I suggested to change the link color. Did you check how the headings are defined in the message html in the frontend?

**Reply by blissiraDC** [reply, marked as answer]
Hey @ccamedia, I found the answer to this. I added this to index.css, change the color of course to whatever you require. 

I have noticed that it sometimes takes a while for the changes to take effect,  maybe that JSdelvr has to propagate web.js?  

Here is the code

.prose :where(strong):not(:where([class~=not-prose],
[class~=not-prose] *)) {
  color:#abe7ff;
  font-weight:600
}

**Reply by ccamedia** [reply]
Hey @blissiraDC, thank you for your quick response. Does this affect all the headlines and bullet points? I've made the changes now. How long did it take for the color to update in your case? It does not look like it works right now... hmm

**Reply by blissiraDC** [reply]
It picks up the bullets and normal font size headlines, but for larger headline fonts like what GPT 4o outputs no, I am still trying to get that showing. Its very weird but I have found that I have to refresh several times or even completely clear my browser cache before I see any changes. Which seems to be the case more often than not.

**Reply by ccamedia** [reply]
Hey @blissiraDC , thanks so much for your help. Unfortunately, the chatbot I've been working on for so long is messing up the CSS on my website, and I'm stuck... It's really frustrating! Most of my clients use the same theme as I do (Divi), so this issue might completely derail my plans to sell personalized AI chatbots. You can see what happens when I embed the chatbot on my test page... [cca-media.at/test/](https://cca-media.at/test/) :(

**Comment by blissiraDC** [comment]
@ccamedia how did you get the gradient color in the header?

**Reply by ccamedia** [reply]
Hey blissiraDC - just copy this css in the "theme": backgroundColor: "linear-gradient(45deg, #8839ae, #db2547, #fc9a3f)", you can adjust the colors as you want.

**Comment by blissiraDC** [comment]
Hey @ccamedia thanks for your reply, unfortunately that option is not available in full page embed. I would love an answer to your question though, it seems nothing I change in index.css seems to have any effect.
